### PR TITLE
Choice for receiving calls, texts and call answered alerts; add call time modal dialog

### PR DESCRIPTION
**Pull request by sharett**
I added the option for receiving calls and receiving call answered alerts.  Call answered alerts will allow situation navigators to receive a text when a call is answered, and know who answered it.

I also went ahead and added a modal dialog for easier adding of call time entries.  I probably should have done this in a separate pull request but I was inspired.

### CODE REVIEW

File: lib/lib_sms.php [hunk 1]
```diff
@@ -25,26 +25,34 @@
 *   Array of contacts to be loaded.
 * @param int $language_id
 *   The language the contact must speak.  If zero, all languages will be returned.
-* @param bool $texting
-*   If true, the contact must support texting.
+* @param array $receives
+*   An array of boolean values, with keys 'calls', 'texts' and 'answered_alerts'.  If
+*   these are set to true, then only contacts that support receiving these will be
+*   returned.
+* @param bool $calls
```
Comment (misplacedpriorities) [inline lib/lib_sms.php]: This parameter listing is erroneous. Was it intended?

**Review by misplacedpriorities — COMMENTED** [review]
Haven't tested yet, not sure if this is even worth mentioning, but there's an extraneous parameter specified in a function header comment.

File: lib/lib_sms.php (hunk 1 above)
Comment (sharett) [reply lib/lib_sms.php]: No, I forgot to remove the $calls parameter.  I'll push a fix.

Comment (sharett) [reply lib/lib_sms.php]: Actually, this was fixed in a later commit.  For some reason we're only viewing changes from 1 commit.  If you select "changes from all commits" then it is fixed!

Comment (misplacedpriorities) [reply lib/lib_sms.php]: Apologies, user error.

**Comment by misplacedpriorities** [comment]
There appears to be a bug with the "Add call time for ..." dialog. It seems to take the parameters I input, except for the start time and end time; regardless of what I've input, it always uses 12:00am to 11:59pm as the time boundaries.

So far, no other bugs encountered. Will continue testing.

**Comment by sharett** [comment]
Thanks Chris - I fixed the bug.  The variable should have called "earliest" but was called "earliest_time".  Same for "latest".

**Comment by misplacedpriorities** [comment]
Tests out successfully now -- a real improvement for adding call times.

One question: It is possible in the "Add call time for..." dialog to not check any of the three options at the bottom (call, text, call answered alerts). Is this what we want? Does it make sense for the user to add a call time with none of those boxes checked?

**Comment by sharett** [comment]
I am not sure.  The situation navigators would be on duty but not receiving calls or texts, but they typically would receive call answered alerts.  Let's ask at the next meeting.

OK to merge this branch?

**Comment by misplacedpriorities** [comment]
Yes, looks OK to me, thanks. I'd be interested in talking about this briefly at the next meeting, as you suggested.

**Pull request merged**